pkg/koordlet/runtimehooks/nri/server.go

```diff
@@ -104,19 +108,34 @@ func NewNriServer(opt Options) (*NriServer, error) {
 }
 
 func (p *NriServer) Start() error {
-	go func() {
+	err := p.options.Validate()
+	if err != nil {
+		return err
+	}
+	success := time.After(p.options.NriConnectTimeout)
```

Comment:
this will make stub.Run always success after `NriConnectTimeout`

Reply:
yes, and once stub.Run successfully, it will never return. So if after this timeout, we couldn't get any response, we should regard it as success.

> this will make stub.Run always success after `NriConnectTimeout`

Reply:
so maybe we don't need to do `errorChan <- err`, and just return error directly?

Reply:
maybe we can't return error directrly here since we should handle the error in other goroutine. And since the error return may need some time in the goroutine, we may need one channel to handle the error which is generated in the other goroutine.